apps/platform-app/src/auth.ts

```diff
@@ -111,17 +137,59 @@ export const {
         throw new Error('Account not found')
       }
       const userIsHunter = userHasRole(foundUser, Role.HUNTER)
-      if (
-        (account.provider === 'github' && !userIsHunter) ||
-        (account.provider === 'email' && userIsHunter)
-      ) {
-        console.log('ACCESS GRANTED')
-        return true
-      } else {
+
+      // Define provider-specific access rules
+      const providerAccess = {
+        email: () => {
+          if (userIsHunter) {
+            return true
+          }
+          throwAccessDeniedError(account.provider, userIsHunter)
+        },
+        github: async () => {
+          if (userIsHunter)
+            throwAccessDeniedError(account.provider, userIsHunter)
+          const githubUsername = profile?.login as string
+          if (!githubUsername)
+            throw new Error(
+              '[GitHub provider] GitHub profile login not provided.',
+            )
+          await createGitHubDetailsForUser(foundUser.id, githubUsername)
+          return true
+        },
+        linkedin: async () => {
+          if (userIsHunter)
```

Comment:
**issue:** We should allow hunters to connect their LinkedIn as well, even if they already have acccount created by magic link